ddtrace/tracer/option.go

```diff
@@ -482,6 +554,43 @@ func (c *config) loadAgentFeatures() {
 	}
 }
 
+// ImportIntegration labels the given integration as imported
+func ImportIntegration(ip string) bool {
+	s, ok := agentIntegrations[ip]
+	if !ok {
+		log.Error("Attempted to import invalid integration %s", ip)
+		return false
+	}
+	s.imported = true
+	agentIntegrations[ip] = s
+	return true
+}
+
+func (c *config) loadAgentIntegrations() {
+	integrations := map[string]integrationConfig{}
+	deps, ok := debug.ReadBuildInfo()
+	for _, s := range agentIntegrations {
+		ic := integrationConfig{
+			Instrumented: s.imported,
+		}
+		if !ok {
+			integrations[s.name] = ic
+			continue
+		}
+		for _, d := range deps.Deps {
+			if d.Path == s.gName {
```

Comment:
Did you test this locally? I'm a bit confused why sometimes the `gName` is a regexp with things like`^` and `$` and sometimes it's just a string?

Reply:
Yup, tested w each of the integrations 😫 . The main issue was the `olivere/elastic` integration -- the DataDog integration can work with _either_ gopkg.in/olivere/elastic.v3 or v5. 

For `^...$`, I was running into problems where something like `go-chi/chi` would match both `go-chi/chi` and `go/chi/v5` since `MatchString` seems to act more like a `Contains` than an "exact match".

But if we implement what I mentioned in the above comment (about the script + using code to parse the integration name from the library name), I think this problem might go away/be easier to get around.

Reply:
Hm, I'm definitely open to seeing if we can reduce this to a single loop, but unsure if I follow what you mentioned. `deps.Deps` is a slice of structs, so the only map here is `contribIntegrations`, but I think trying to access from it using keys will introduce the same naming issues mentioned in the above comment thread :O Is there something else you had in mind?

Reply:
[resolved] :)